src/arbitraries/ArbitraryInteger.ts

```diff
@@ -28,14 +28,14 @@ export class ArbitraryInteger extends Arbitrary<number> {
       const lower = Math.max(0, this.min)
       const upper = Math.max(lower, initial.value - 1)
 
-      if (lower === upper) return NoArbitrary
+      if (lower === upper && initial.value !== 1) return NoArbitrary
```

Comment:
Are you sure this will work for any integer interval that doesn't overlap 0?

Reply:
Do you mean like [5,10] for example?

Reply:
I confirmed it also works. I used this particular test for making the verification

```
it('finds that summing two positive numbers in a range nevers returns zero', () => {
    expect(fc.scenario()
      .config(fc.strategy()
        .withRandomSampling()
        .usingCache()
        .withoutReplacement()
        .withShrinking()
      )
      .forall('a', fc.integer(5, 10))
      .exists('b', fc.integer(1, 2))
      .then(({a, b}) => a + b === 0)
      .check()
    ).to.have.property('satisfiable', false)
  })
```

I think the modification is safe because it uses directly the `initial.value` and not the `upper` and `lower` limits.

Reply:
Let me know if have any more doubts!

Reply:
I am sorry, but I am still not conviced this actually works for all cases 😅 @arestivo, @ruippeixotog could you do a sanity check here? What makes `-1` and `1` special when shrinking an `fc.integer(5, 10)`?

Reply:
There's nothing special about `-1` and `1`, I think this PR just special cases the shrinker for `fc.integer(0, n)` ranges, where it will now shrink to 0. My guess is that if you try to shrink some value in `fc.integer(5, 10)` it will still shrink to 6, not 5 as intended.

Without compiling or testing anything, the real problem seems to be the `lower === upper` part. In those cases we are incorrectly returning `NoArbitrary` (which means that we didn't find any "smaller" input). The condition should be changed to `lower > upper` and `upper` should be defined simply as `initial.value - 1`.

Reply:
`-1` and `1` were only being _special_ due to the way code was developed, which did not allow to shrink from either `-1` or `1` to `0`.

However, I completely agree with @ruippeixotog's last comment since the _suggested_ modifications make total sense. I tried them and they work. Already pushed a new version of fixing the `ArbitraryInteger` shrinking problem. @hugoferreira  Let me know how you feel about the new solution now.